Overview.bs

```diff
@@ -212,39 +212,129 @@ Opt-In Mechanism {#opt-in}
-3. If the client receives the patch-subset magic number, it commences using the patch-subset method. Otherwise, if the client receives the <code>Accept-Ranges: bytes</code> header, it commences using the range-request method. Otherwise, the whole font file is downloaded, and the current non-incremental loading behavior is used.
+Note: the most recently specified method by the content takes precedence. For example if on a site the
+first page specifies to use the patch subset method for a font URL, but the second page specifies
+range request for the same font URL then the second page should load the font using range request.
+Such cases will require the client to reset any previously stored state for that URL and start fresh
+with the new method.
+
+
+IFT Method Negotiation {#method-negotiation}
+--------------------------------------------
+
+When the particular IFT method(s) that are supported by a server are not known, the client must
+determine which method to use. Different clients may support different IFT methods, and different
+servers may support different IFT methods, so a negotation occurs as such:
+
+1.  The browser makes the first request to the server using the GET HTTP method ([[rfc9110#name-get]]).
+     If the client prefers the <a href="#patch-incxfer">patch-subset method</a>, it sends the relevant
+     [=patch request header=]. If the client prefers the range-request method, it does not send the
+     header.
+
+2.  If the server receives the patch request header and wishes to honor it, the server must reply with
+     a valid <a href="#PatchResponse">patch subset response</a> which includes the
+     <a href="#handling-patch-response">patch-subset magic number</a>. Otherwise, the server must
+     reply with the [[RFC9110#range.requests]]
+     <a href="https://www.rfc-editor.org/rfc/rfc9110.html#name-accept-ranges"><code>Accept-Ranges</code></a>
+     header, if it supports HTTP Range Requests.
+     
+3.  If the client receives the patch-subset magic number, it commences using the patch-subset method.
+     Otherwise, if the client receives the <code>Accept-Ranges: bytes</code> header, it commences
+     using the range-request method. Otherwise, the whole font file is downloaded, and the current
+     non-incremental loading behavior is used.
 
 IFT Method Fallback {#fallback}
 -------------------------------
 
 <em>This section is not normative.</em>
 
-This summarizes behaviors that result from the above method selection.
+This summarizes behaviors that result from the above method selection and negotiation sections.
```

Comment:
In the past, when we talked about whether we should mandate both methods, one of the arguments was that while it would be trivial for a patch-subset server to also support range-request, it may not be the case for a server that only supports range-request. The proposed method selection implies that either the author/content method selection is fulfilled, or the non-incremental approach is used, regardless of any other considerations. Should we revisit the earlier discussions to determine the possible outcomes?

Reply:
Patch subset servers supporting range request is still relevant to the incremental-auto case. For the method specific keywords (incremental-range/patch) I considered allowing incremental-patch to fallback to range if the client only supports range, but my thought is that the content has very explicitly asked for a specific method so we should only use the method that's been requested.

If the content is OK with either method it can declare a font face with two urls one tagged with incremental-patch and the other with incremental-range (or incremental-auto). Then it would be up to the client to pick which one to use.

Reply:
Makes sense. thank you for clarification.